drivers/ng_at86rf2xx/ng_at86rf2xx_netdev.c

```diff
@@ -0,0 +1,676 @@
+            }
+            break;
+        case NETCONF_STATE_RESET:
+            ng_at86rf2xx_reset(dev);
+            break;
+        default:
+            return -ENOTSUP;
+    }
+    return sizeof(ng_netconf_state_t);
+}
+
+ng_netconf_state_t _get_state(ng_at86rf2xx_t *dev)
+{
+    switch (ng_at86rf2xx_get_state(dev)) {
+        case NG_AT86RF2XX_STATE_SLEEP:
+            return NETCONF_STATE_SLEEP;
+        case NG_AT86RF2XX_STATE_BUSY_RX_AACK:
+            return NETCONF_STATE_RX;
+        case NG_AT86RF2XX_STATE_BUSY_TX_ARET:
+        case NG_AT86RF2XX_STATE_TX_ARET_ON:
+            return NETCONF_STATE_TX;
+        case NG_AT86RF2XX_STATE_RX_AACK_ON:
+        default:
+            return NETCONF_STATE_IDLE;
+    }
+}
+
+static int _get(ng_netdev_t *device, ng_netconf_opt_t opt,
+                void *val, size_t max_len)
+{
+    if (device == NULL) {
+        return -ENODEV;
+    }
+    ng_at86rf2xx_t *dev = (ng_at86rf2xx_t *) device;
+
+    switch (opt) {
+
+        case NETCONF_OPT_ADDRESS:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = ng_at86rf2xx_get_addr_short(dev);
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_ADDRESS_LONG:
+            if (max_len < sizeof(uint64_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint64_t *)val) = ng_at86rf2xx_get_addr_long(dev);
+            return sizeof(uint64_t);
+
+        case NETCONF_OPT_ADDR_LEN:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = 2;
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_SRC_LEN:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            if (dev->options & NG_AT86RF2XX_OPT_SRC_ADDR_LONG) {
+                *((uint16_t *)val) = 8;
+            }
+            else {
+                *((uint16_t *)val) = 2;
+            }
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_NID:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = dev->pan;
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_CHANNEL:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            ((uint8_t *)val)[1] = 0;
+            ((uint8_t *)val)[0] = ng_at86rf2xx_get_chan(dev);
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_TX_POWER:
+            if (max_len < sizeof(int16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = ng_at86rf2xx_get_txpower(dev);
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_MAX_PACKET_SIZE:
+            if (max_len < sizeof(int16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = NG_AT86RF2XX_MAX_PKT_LENGTH;
```

Comment:
You are returning the full 802.15.4 packet length here without taking care of the 802.15.4 header that also has to be appended.